Source/Plugins/Core/com.equella.core/js/tsrc/modules/GallerySearchModule.ts

```
@@ -0,0 +1,303 @@
+        E.fromNullable("Attachment is missing its `filePath`"),
+        E.map((filePath) =>
+          buildFileAttachmentUrl(itemUuid, itemVersion, filePath)
+        )
+      )
+    )
+    .done();
+
+/**
+ * Creates a `mainEntry` for a `GalleryEntry` based on an item's attachments. This specifically
+ * entails using the very first attachment - or returning an error (E.Left) if there isn't one.
+ *
+ * @param itemUuid The UUID of the item for which this entry is for
+ * @param itemVersion The version of the item for which this entry is for
+ * @param attachments An item's attachments
+ */
+const createMainEntry = (
+  itemUuid: string,
+  itemVersion: number,
+  attachments: OEQ.Search.Attachment[]
+): E.Either<string, GalleryEntry> =>
+  pipe(
+    attachments,
+    A.head,
+    E.fromOption(() => "Missing attachment for main entry"),
+    E.chain((attachment) =>
+      buildGalleryEntry(itemUuid, itemVersion, attachment)
+    )
+  );
+
+/**
+ * Similar to `createMainEntry` in intent, but operating on the ideally a provided `A.tail` of
+ * attachments. Further, is gentler with errors (preferring to return an O.Option) as additional
+ * entries are kind of nice to have.
+ *
+ * @param itemUuid The UUID of the item for which this entry is for
+ * @param itemVersion The version of the item for which this entry is for
+ * @param attachments Should be the `A.tail` of an item's attachments to potentially make up
+ *                    additional entries.
+ */
+const createAdditionalEntries = (
+  itemUuid: string,
+  itemVersion: number,
+  attachments: O.Option<OEQ.Search.Attachment[]>
+): O.Option<GalleryEntry[]> =>
+  pipe(
+    attachments,
+    O.map(
+      flow(
+        A.map((attachment) =>
+          pipe(
+            buildGalleryEntry(itemUuid, itemVersion, attachment),
+            // Let's log any issues - just so we're aware of any odd data
+            E.mapLeft((error) => {
+              console.warn(error);
+              return error;
```

Do we need to return the error ?

---

Best to, as we're still in an Either and the intent here is just to log any issues. I acknowledge that down below we do then strip stuff, but that knowledge should not be coupled to here.

---

Addressed also via the refactoring.